data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/export/DataFileLoader.java

```diff
@@ -170,11 +194,15 @@ public void run() {
             }
 
             lines.clear();
+
             reader.close();
+            gzipInputStream.close();
+            inputStream.close();
             LOG.info("Complete loading s3://{}/{}", bucketName, key);
-        } catch (Exception e) {
+        } catch (IOException e) {
```

Comment:
Why is this exception scoped down ? How are we handling other exception ?

Reply:
That is true. There may have other exceptions. Since this is merged, I will update this in the next PR.